src/Libraries/RevitNodes/Elements/CurtainSystem.cs

```diff
@@ -0,0 +1,209 @@
+﻿using System;
+using System.Linq;
+using Autodesk.DesignScript.Geometry;
+using Autodesk.Revit.DB;
+using Revit.GeometryConversion;
+using RevitServices.Persistence;
+using RevitServices.Transactions;
+using Curve = Autodesk.DesignScript.Geometry.Curve;
+using System.Collections.Generic;
+using Autodesk.DesignScript.Runtime;
+
+namespace Revit.Elements
+{
+    /// <summary>
+    /// A Revit CurtainSystem
+    /// </summary>
+    public class CurtainSystem : Element
+    {
+        #region Internal properties
+
+        /// <summary>
+        /// An internal handle on the Revit CurtainSystem
+        /// </summary>
+        internal Autodesk.Revit.DB.CurtainSystem InternalCurtainSystem
+        {
+            get; private set;
+        }
+
+        /// <summary>
+        /// Reference to the Element
+        /// </summary>
+        [SupressImportIntoVM]
+        public override Autodesk.Revit.DB.Element InternalElement
+        {
+            get { return InternalCurtainSystem; }
+        }
+
+        #endregion
+
+        #region Private constructors
+
+        /// <summary>
+        /// Private constructor
+        /// </summary>
+        private CurtainSystem(Autodesk.Revit.DB.CurtainSystem curtainSystem)
+        {
+            SafeInit(() => InitCurtainSystem(curtainSystem));
+        }
+      
+        /// <summary>
+        /// Private constructor
+        /// </summary>
+        private CurtainSystem(ReferenceArray faces, Autodesk.Revit.DB.CurtainSystemType curtainSystemType)
+        {
+            SafeInit(() => InitCurtainSystem(faces, curtainSystemType));
+        }
+
+        #endregion
+
+        #region Helpers for private constructors
+
+        /// <summary>
+        /// Initialize a CurtainSystem element
+        /// </summary>
+        private void InitCurtainSystem(Autodesk.Revit.DB.CurtainSystem curtainSystem)
+        {
+            InternalSetCurtainSystem(curtainSystem);
+        }
+
+        /// <summary>
+        /// Initialize a CurtainSystem element
+        /// </summary>
+        private void InitCurtainSystem(ReferenceArray faces, Autodesk.Revit.DB.CurtainSystemType curtainSystemType)
```

Comment:
Yes right, will change to accept only one surface and then bind to the first result of NewCurteinSystem2